### PR TITLE
Suisin/chore: implement session expired timer

### PR DESCRIPTION
**Pull request by suisin-deriv**
## Changes:

Implement session timeout modal

### Screenshots:

Please provide some screenshots of the change.

### CODE REVIEW

File: packages/account/src/Sections/Profile/PhoneVerification/cancel-phone-verification-modal.tsx [hunk 1]
```diff
@@ -17,10 +17,11 @@ const CancelPhoneVerificationModal = observer(() => {
     const { setVerificationCode, is_virtual } = client;
     const { isMobile } = useDevice();
     const { trackPhoneVerificationEvents } = usePhoneVerificationAnalytics();
+    const { should_show_session_timeout_modal: is_session_exired } = usePhoneNumberVerificationSessionTimer();
 
     useEffect(() => {
         const unblock = history.block((location: Location) => {
-            if (!show_modal && !is_virtual && !is_forced_to_exit_pnv) {
+            if (!show_modal && !is_virtual && !is_session_exired && !is_forced_to_exit_pnv) {
```
Comment (matin-deriv) [inline packages/account/src/Sections/Profile/PhoneVerification/cancel-phone-verification-modal.tsx:24]:
```suggestion
            if (!show_modal && !is_virtual && !is_session_expired && !is_forced_to_exit_pnv) {
```

File: packages/hooks/src/usePhoneNumberVerificationSessionTimer.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    const { account_settings } = client;
+    const { phone_number_verification } = account_settings;
+
+    useEffect(() => {
+        WS.send({ time: 1 }).then((response: { error?: Error; time: number }) => {
+            if (response.error) return;
+
+            if (response.time && phone_number_verification?.session_timestamp) {
+                const request_in_milliseconds = dayjs(phone_number_verification?.session_timestamp * 1000);
+                const next_request = Math.round(request_in_milliseconds.diff(response.time * 1000) / 1000);
```
Comment (matin-deriv) [inline packages/hooks/src/usePhoneNumberVerificationSessionTimer.ts:19]: Leave comments for these 2 lines to explain the functionality and the time for better understanding.

File: packages/account/src/Sections/Profile/PhoneVerification/cancel-phone-verification-modal.tsx [hunk 3]
```diff
@@ -29,7 +30,7 @@ const CancelPhoneVerificationModal = observer(() => {
         });
 
         return () => unblock();
-    }, [history, show_modal, is_virtual, is_forced_to_exit_pnv]);
+    }, [history, show_modal, is_virtual, is_session_exired, is_forced_to_exit_pnv]);
```
Comment (fasihali-deriv) [inline packages/account/src/Sections/Profile/PhoneVerification/cancel-phone-verification-modal.tsx:33]: Here as well `is_session_exired`

**Review by fasihali-deriv — APPROVED** [review]

**Review by matin-deriv — APPROVED** [review]

File: packages/account/src/Sections/Profile/PhoneVerification/session-timeout-modal.tsx [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+        history.push(routes.personal_details);
+    };
+
+    return (
+        <Modal
+            isMobile={isMobile}
+            isOpened={should_show_session_timeout_modal}
+            primaryButtonCallback={redirectBackToPersonalDetails}
+            primaryButtonLabel={<Localize i18n_default_text='OK' />}
+            title='Session Expired'
```
Comment (amina-deriv) [inline packages/account/src/Sections/Profile/PhoneVerification/session-timeout-modal.tsx:23]: should we localize the title?

**Review by amina-deriv — APPROVED** [review]

**Pull request merged**